types/mapcmd/map.go

```diff
@@ -109,17 +109,22 @@ func decorateCommandWithMapNameFlags(cmd *cobra.Command, mapName *string) {
 	cmd.MarkFlagRequired("name")
 }
 
-func decorateCommandWithKeyFlags(cmd *cobra.Command, mapKey *string) {
-	cmd.Flags().StringVarP(mapKey, "key", "k", "", "key of the map")
+func decorateCommandWithKeyFlags(cmd *cobra.Command, mapKey, mapKeyType *string) {
+	flags := cmd.Flags()
+	flags.StringVarP(mapKey, "key", "k", "", "key of the map")
+	flags.StringVarP(mapKeyType, "key-type", "", "string", "type of the key, one of: string, json, int(8,16,32,64), float(32,64)")
 	cmd.MarkFlagRequired("key")
+	cmd.RegisterFlagCompletionFunc("key-type", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```

Comment:
If this should autocomplete in the prompt, I was not able to trigger it:

![image](https://user-images.githubusercontent.com/32355782/171357374-60dd497e-3472-43cf-9fad-79a6b8ad1d93.png)

Is it because I build from source code?